src/modules/project-config.ts

```diff
@@ -256,3 +269,75 @@ export class ProjectConfig {
         return VariableCollection.build(this.getComponents(), this.getVariableMappings(), componentContext);
     }
 }
+
+export class ProjectConfigLock {
+    packages: PackageConfig[];
+
+    constructor(packages: PackageConfig[]) {
+        this.packages = packages;
+    }
+
+    static isAvailable = (path: PathLike = DEFAULT_CONFIG_LOCKFILE_PATH) => CliFileSystem.existsSync(path);
+
+    static read(path: PathLike = DEFAULT_CONFIG_LOCKFILE_PATH): ProjectConfigLock | null {
+        try {
+            const data = JSON.parse(CliFileSystem.readFileSync(path as string));
+            const packages = data.packages;
+            return new ProjectConfigLock(packages);
+        } catch {
+            return null;
+        }
+    }
+
+    /**
+     * Writes the locked project configuration to file.
+     * @param projectConfig Project configuration to get the packages for the lock file.
+     * @param path Path of the file to write the configuration to. Defaults to DEFAULT_CONFIG_LOCKFILE_PATH.
+     */
+    static write(projectConfig: ProjectConfig, path: PathLike = DEFAULT_CONFIG_LOCKFILE_PATH): void {
+        try {
+            const projectConfigOptions = {
+                packages: projectConfig.getPackages(),
+            };
+            const configString = JSON.stringify(projectConfigOptions, null, 4);
+            CliFileSystem.writeFileSync(path, configString);
+        } catch (error) {
+            throw new Error(`Error writing .velocitas-lock.json: ${error}`);
+        }
+    }
+
+    /**
+     * Updates the lock file with the new version of a package.
+     * @param packageConfig Package configuration with updated version information.
+     * @param path Path of the file to update. Defaults to DEFAULT_CONFIG_LOCKFILE_PATH.
+     */
+    static update(packageConfig: PackageConfig, path: PathLike = DEFAULT_CONFIG_LOCKFILE_PATH): void {
```

Comment:
Is a single package update method really required? We are re-writing the complete file anyway, cant we just call read/write?